backend/lib/endpoints/posts.js

```diff
@@ -374,42 +374,121 @@ async function routes(app) {
     },
   );
 
-  app.post(
+  // -- Comments
+  const COMMENT_PAGE_SIZE = 5;
+
+  app.get(
     "/:postId/comments",
-    { preValidation: [app.authenticate], schema: addCommentSchema },
+    { preValidation: [app.authenticate], schema: getCommentsSchema },
     async (req) => {
-      const { body, params, userId } = req;
-      const { parentId } = body;
+      const { limit, skip } = req.query;
+      const { postId } = req.params;
+      const [commentErr, comments] = await app.to(
+        Comment.aggregate([
+          {
+            $match: {
+              parentId: null,
+              postId: mongoose.Types.ObjectId(postId),
+            },
+          },
+          {
+            $lookup: {
+              as: "children",
+              foreignField: "parentId",
+              from: "comments",
+              localField: "_id",
+            },
+          },
+          {
+            $addFields: {
+              childCount: {
+                $size: { $ifNull: ["$children", []] },
+              },
+            },
+          },
+          {
+            $skip: skip || 0,
+          },
+          {
+            $limit: limit || COMMENT_PAGE_SIZE,
+          },
+        ]),
+      );
+      if (commentErr) {
+        req.log.error("Failed retrieving comments", { commentErr });
+        throw app.httpErrors.internalServerError();
+      }
+
+      return comments;
+    },
+  );
+
+  app.post(
+    "/:postId/comments",
+    { preValidation: [app.authenticate], schema: createCommentSchema },
+    async (req, reply) => {
+      const { userId, body: commentProps, params } = req;
       const { postId } = params;
-      if (parentId) {
-        const parentPost = await Post.findById(parentId);
-        if (!parentPost || parentPost.postId !== postId) {
-          return new httpErrors.BadRequest();
-        }
+
+      const [userErr, user] = await app.to(User.findById(userId));
+      if (userErr) {
+        req.log.error("Failed retrieving user", { userErr });
```

Comment:
Sorry for the delay in reviewing but wanted to get caught up in posts/users/orgs first. Anyways these scripts like `const [userErr, user] = await app.to(User.findById(userId));`, If the query is legit and nothing else errors out but the user doesn't exist for whatever reason - both `userErr` and `user` are `null`. Should we also include `if ( userErr || user === null)` instead of just `if (userErr)`?

Reply:
@joshmorel since you've added a separate PR to fix these issues can we approve this now?